examples/cpp/helloworld/mtls_greeter_server.cc

```diff
@@ -0,0 +1,109 @@
+/*
+ *
+ * Copyright 2020 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include <fstream>
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <grpcpp/ext/proto_server_reflection_plugin.h>
+#include <grpcpp/grpcpp.h>
+#include <grpcpp/health_check_service_interface.h>
+
+#include "absl/flags/flag.h"
+#include "absl/flags/parse.h"
+#include "examples/protos/helloworld.grpc.pb.h"
+
+using grpc::Server;
+using grpc::ServerBuilder;
+using grpc::ServerContext;
+using grpc::Status;
+using helloworld::Greeter;
+using helloworld::HelloReply;
+using helloworld::HelloRequest;
+
+ABSL_FLAG(std::string, port, "50051",
+          "The port number to use for the gRPC connection.");
+ABSL_FLAG(std::string, client_root_cert_pem_path, "ca.cert",
+          "The path to the root X509 certificate.");
+ABSL_FLAG(std::string, server_cert_pem_path, "service.pem",
+          "The path to the server's X509 certificate.");
+ABSL_FLAG(std::string, server_key_pem_path, "service.key",
+          "The path to the server's private key.");
+
+// Logic and data behind the server's behavior.
+class GreeterServiceImpl final : public Greeter::Service {
+  Status SayHello(ServerContext* context, const HelloRequest* request,
+                  HelloReply* reply) override {
+    if ((!reply) || (!request)) {
+      return Status::CANCELLED;
+    }
+    std::string prefix("Hello ");
+    reply->set_message(prefix + request->name());
+    std::cout << "Received message: " << request->name() << std::endl;
+    return Status::OK;
+  }
+};
+
+static std::string readFile(const std::string& filePath) {
+  std::ifstream ifs(filePath);
+  return std::string((std::istreambuf_iterator<char>(ifs)),
+                     (std::istreambuf_iterator<char>()));
+}
+
+void RunServer() {
+  std::string port = absl::GetFlag(FLAGS_port);
```

Comment:
Please make this into the full server address.

Reply:
Shouldn't the server only have it's port be configurable? Since the server will always have it's address as localhost + the port number.

Reply:
Since this is what we also do for Go, let's leave this as is.